### PR TITLE
Add some dual N/P MOSFET devices

### PR DESCRIPTION
**Pull request by dbrgn**
Generated with https://github.com/LibrePCB/librepcb-parts-generator/pull/34

### CODE REVIEW

**Comment by ubruhin** [comment]
Hmm it seems that many of them are exactly identically, except their name. So couldn't we useplaceholders (like "x") in the package name to reduce the amount of devices? See for example https://github.com/LibrePCB-Libraries/Microchip.lplib/pull/6.

**Comment by dbrgn** [comment]
Most of the parts have different specs though, even if the signals and package are identical. Wasn't the concept always "a device is a concrete component that can be purchased"?

I think it's fine to use placeholders for devices that come in different variants (e.g. lead-free and leaded, or 512K of RAM vs 256K of RAM). These case will only vary in configuration, so it can be treated like the resistance or tolerance of a restistor. In this PR, there are a few devices that look very similar (DMC3025LSDQ PPAP capable while DMC3025LSD isn't), but I don't know what PPAP is and whether it could have any influence on the circuit design.

If multiple parts with their own datasheets share the same device, we have to make sure that they are identical and exchangeable in all circuits. I don't know if there's a good way to do that unless you know the entire product range really well.

Another aspect is the placeholder name: If the fictional parts DMC3033LGA and DMC3033LGB exist, and we create a common DMC3033LGx device for them, how do we make sure that a DMC3033LGC doesn't show up a year later with a different plastic body that would require a different 3D model?

I would prefer to err on the side of adding too many devices. If a device maps to a single orderable item it also simplifies BOM handling. I think "if the device has its own datasheet it should be a separate device" could be a good indicator. (Although a single datasheet might even result in multiple devices, e.g. if there are different body colors available.)

**Comment by ubruhin** [comment]
I see your point, but generally a device is just the connection between a component and a package. If 100 parts represent the same component, have the same package and the same pinout, they could all share the same device. This way we can avoid defining many identical devices (to reduce duplication, effort and the risk of making mistakes in the pinout).

But because it would be strange to create devices like "NMOS SOT23-3 D=1,G=2,S=3" we create separate devices per parts family, just because the devices can then have much more expressive names and are thus easier to understand.

So in the end, it's a compromise. But currently devices are not meant to represent MPNs, it's still an open question how we should handle MPNs.

> Another aspect is the placeholder name: If the fictional parts DMC3033LGA and DMC3033LGB exist, and we create a common DMC3033LGx device for them, how do we make sure that a DMC3033LGC doesn't show up a year later with a different plastic body that would require a different 3D model?

Sure, there is always a little risk of having wrong device names because of the placeholders. But I think the risk is very low, and we could fix it at any time, for example by renaming to "DMC3033LGA/B".

**Comment by dbrgn** [comment]
I updated the PR at https://github.com/LibrePCB/librepcb-parts-generator/pull/34.

**Comment by dbrgn** [comment]
@ubruhin PR is updated with the changes from [LibrePCB/librepcb-parts-generator#34](https://github.com/LibrePCB/librepcb-parts-generator/pull/34).

File: dev/3fb84ce0-605c-4dc6-a2c6-a85a1441f5da/device.lp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+(librepcb_device 3fb84ce0-605c-4dc6-a2c6-a85a1441f5da
+ (name "DMC3071LVT")
+ (description "Diodes Incorporated DMC3071LVT Dual MOSFET N/P-Channel 30V.\n\nDatasheet:\n- h\n- t\n- t\n- p\n- s\n- :\n- /\n- /\n- w\n- w\n- w\n- .\n- d\n- i\n- o\n- d\n- e\n- s\n- .\n- c\n- o\n- m\n- /\n- a\n- s\n- s\n- e\n- t\n- s\n- /\n- D\n- a\n- t\n- a\n- s\n- h\n- e\n- e\n- t\n- s\n- /\n- D\n- M\n- C\n- 3\n- 0\n- 7\n- 1\n- L\n- V\n- T\n- .\n- p\n- d\n- f\n\nGenerated with librepcb-parts-generator (generate_mosfet_dual.py)")
```
Comment (ubruhin) [inline dev/3fb84ce0-605c-4dc6-a2c6-a85a1441f5da/device.lp]: Strange datasheet URL :grin:

Comment (dbrgn) [reply dev/3fb84ce0-605c-4dc6-a2c6-a85a1441f5da/device.lp]: Oh I'm such an idiot... :joy: :man_facepalming:

Comment (dbrgn) [reply dev/3fb84ce0-605c-4dc6-a2c6-a85a1441f5da/device.lp]: Should be fixed.

**Review by ubruhin — APPROVED** [review]
:+1:

**Pull request merged**